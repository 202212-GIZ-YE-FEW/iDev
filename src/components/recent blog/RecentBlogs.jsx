import { withTranslation } from "next-i18next";

import BlogItem from "./BlogItem";
import PageIntro from "../PageIntro";
function RecentBlogs({ t }) {
    const recentBlogs = [
        {
            title: t("blog1Title"),
            thumbnail: "blog1",
        },
        {
            title: t("blog2Title"),
            thumbnail: "blog2",
        },
    ];
    return (
        <>
            <div className='container flex flex-col space-y-20 justify-between'>
                <PageIntro title={t("recentBlogsTitle")} />
                <div className='grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5 justify-items-center'>
                    {recentBlogs.map((item, index) => {
                        return (
                            <>
                                <BlogItem
                                    key={item.key}
                                    title={item.title}
                                    thumbnail={item.thumbnail}
                                    isOdd={index % 2 === 0 ? false : true}
                                />
                                <BlogItem
                                    key={item.key}
                                    title={item.title}
                                    thumbnail={item.thumbnail}
                                    isOdd={index % 2 === 0 ? false : true}
                                />
                            </>
                        );
                    })}
                </div>
            </div>
        </>
    );
}
export default withTranslation("home")(RecentBlogs);
