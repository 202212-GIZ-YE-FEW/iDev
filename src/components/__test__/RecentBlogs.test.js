import renderer from "react-test-renderer";

import RecentBlogs from "../recent blog/RecentBlogs";

it("renders correctly", () => {
    const tree = renderer.create(<RecentBlogs />).toJSON();
    expect(tree).toMatchSnapshot();
});
