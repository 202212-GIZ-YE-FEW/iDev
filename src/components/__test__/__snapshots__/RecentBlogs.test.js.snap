// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container flex flex-col space-y-20 justify-between"
>
  <span
    className="font-normal text-3xl md:text-4xl lg:text-5xl block uppercase"
  >
    recentBlogsTitle
  </span>
  <div
    className="react-multi-carousel-list  "
    dir="ltr"
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;
